code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -164,6 +164,7 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 		ASSERT_GAS(giver_id, src)
 		cached_gases[giver_id][MOLES] += giver_gases[giver_id][MOLES]
 
+	SEND_SIGNAL(src, COMSIG_GASMIX_MERGING)
```

Comment:
Some profiles on how much time this takes would be appreciated. If it's too high it may be better to do things in a subtype or something

Reply:
![image](https://user-images.githubusercontent.com/54709710/143677707-7d2f3fe4-53fd-4d04-9e0d-cbd96e90f39c.png)